### ISSUE TITLE
PhantomJS hangs on some "use strict" JS code

### ISSUE THREAD

**Issue by FilippoPolo** [ariya/phantomjs#15046]
Running PhantomJS v. 2.1.1 (binary) on Windows 10. The following code causes PhantomJS to hang indefinitely:

```
console.log("LOADING");
function c() { };
function f() {
    "use strict";
    c({
        t: function () {
            if (true) {
                function s() { }
            }
        }
    });
};
f();
if (typeof phantom == "undefined")
    phantom = { exit: function () { } }
phantom.exit();

```
The code above is valid and will run with no errors on Chrome. In PhantomJS, it will hang, without even displaying the LOADING log line. It still hangs if you remove the call to f(), making it a function that's never invoked.

The problem can be further simplified in this format:

```
console.log("LOADING");
function f() {
    "use strict";
    c({
        t: function () {
            if (true)
                function s() { }
        }
    });
};
phantom.exit();

```
The code above is not valid, and I would expect it to throw a syntax error. That's what it does on Chrome. In PhantomJS, it hangs indefinitely.

Note that if you remove "use strict", then it works.

**Comment by SmaugTheGreat** [comment]
Why did this get closed? It's a pretty severe bug in phantomjs. In general this means that phantomjs is incompatible to any webapps that are using gwt framework, right? At least I couldn't access vudu.com due to this issue.

**Comment by pixiuPL** [comment]
@SmaugTheGreat it got closed as it concerns 2.1.1 version which is **unsupported**
